src/test/java/com/gradle/UtilsTest.java

```diff
@@ -1,22 +1,64 @@
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 import static com.gradle.Utils.toWebRepoUri;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 public class UtilsTest {
-    @Test
-    public void toWebRepoUri_git() {
-        assertEquals(URI.create("https://github.com/acme-inc/my-project"), toWebRepoUri("git://github.com/acme-inc/my-project.git").get());
-        assertEquals(URI.create("https://gitlab.com/acme-inc/my-project"), toWebRepoUri("git://gitlab.com/acme-inc/my-project.git").get());
+
+    @ParameterizedTest
+    @ArgumentsSource(WebRepoUriArgumentsProvider.class)
+    public void testToWebRepoUri(String repositoryHost, String repositoryUri) {
+        URI expectedWebRepoUri = URI.create(String.format("https://%s.com/acme-inc/my-project", repositoryHost));
+        assertEquals(Optional.of(expectedWebRepoUri), toWebRepoUri(String.format(repositoryUri, repositoryHost)));
+    }
+
+    @ParameterizedTest
+    @ArgumentsSource(EnterpriseWebRepoUriArgumentsProvider.class)
+    public void testToWebRepoUri_enterpriseUri(String repositoryHost, String repositoryUri) {
+        URI expectedWebRepoUri = URI.create(String.format("https://%s.acme.com/acme-inc/my-project", repositoryHost));
+        assertEquals(Optional.of(expectedWebRepoUri), toWebRepoUri(String.format(repositoryUri, repositoryHost)));
     }
 
-    @Test
-    public void toWebRepoUri_https() {
-        assertEquals(URI.create("https://github.com/acme-inc/my-project"), toWebRepoUri("https://github.com/acme-inc/my-project").get());
-        assertEquals(URI.create("https://gitlab.com/acme-inc/my-project"), toWebRepoUri("https://gitlab.com/acme-inc/my-project").get());
+    private static class WebRepoUriArgumentsProvider implements ArgumentsProvider {
+
+        @Override
+        public Stream<? extends Arguments> provideArguments(ExtensionContext context) {
+            Set<String> host = Stream.of("github", "gitlab").collect(Collectors.toSet());
+            Set<String> remoteRepositoryUris = Stream.of(
+                    "https://%s.com/acme-inc/my-project",
+                    "https://%s.com:443/acme-inc/my-project",
+                    "https://user:secret@%s.com/acme-inc/my-project",
+                    "ssh://git@%s.com/acme-inc/my-project.git",
+                    "ssh://git@%s.com:22/acme-inc/my-project.git",
+                    "git://%s.com/acme-inc/my-project.git",
+                    "git@%s.com/acme-inc/my-project.git"
+            ).collect(Collectors.toSet());
+            return host.stream().flatMap(h -> remoteRepositoryUris.stream().map(r -> Arguments.arguments(h, r)));
+        }
+    }
+
+    private static class EnterpriseWebRepoUriArgumentsProvider implements ArgumentsProvider {
+
+        @Override
+        public Stream<? extends Arguments> provideArguments(ExtensionContext context) {
+            Set<String> host = Stream.of("github", "gitlab").collect(Collectors.toSet());
```

Comment:
Very nice, don't want to be annoying but we could factorize the `Stream` creation and have only the `Set` of `remoteRepositoryUris` being different in the 2 `ArgumentsProvider` implementations.

Reply:
No worries! So I gave it a shot, and I think I prefer it as-is, since what's being duplicated is very small, and only in two places. If further related argument providers were added, I'd probably be in favor of de-duplicating that.